tools/objc/libtool_check_unique.cc

```diff
@@ -0,0 +1,57 @@
+#include <iostream>
+#include <regex>
+#include <fstream>
+#include <unordered_set>
+
+using namespace std;
+
+string getBasename(const string &path) {
+    // Assumes we're on an OS with "/" as the path separator
+    auto idx = path.find_last_of("/");
+    if (idx == string::npos) {
+        return path;
+    }
+    return path.substr(idx + 1);
+}
+
+// Returns 0 if there are no duplicate basenames in the object files (both via -filelist as well as shell args),
+// 1 otherwise
+int main(int argc, const char * argv[]) {
+    unordered_set<string> basenames;
+    const regex libRegex = regex(".*\\.a$");
+    const regex noArgFlags = regex("-static|-s|-a|-c|-L|-T|-no_warning_for_no_symbols");
```

Comment:
`-D` was added in the other PR, so maybe handle that as well.

```suggestion
    const regex noArgFlags = regex("-static|-s|-a|-c|-L|-T|-D|-no_warning_for_no_symbols");
```

Reply:
added